ksql-engine/src/main/java/io/confluent/ksql/function/udf/math/RoundKudf.java

```diff
@@ -20,11 +20,25 @@
 
 public class RoundKudf implements Kudf {
 
+  public static final String NAME = "ROUND";
+
   @Override
   public Object evaluate(final Object... args) {
-    if (args.length != 1) {
-      throw new KsqlFunctionException("Len udf should have one input argument.");
+    if (args.length != 1 && args.length != 2) {
+      throw new KsqlFunctionException("Round udf should have one or two input arguments.");
+    }
+
+    if (args[0] == null) {
+      return null;
     }
-    return Math.round((Double) args[0]);
+    final Object value = args[0];
+    final Double number = (Double) value;
+
+    if (args.length == 1 || args[1] == null) {
+      return Math.round(number);
+    }
+
+    final Double round = Math.pow(10, (Integer) args[1]);
```

Comment:
Unfortunately this won't work in all cases due the way floating point numbers (e.g. double) represent numbers.
Consider the following:
```
double mult = 265.335d * 100d;
System.out.println(mult);
double res = Math.round(mult) / 100d;
System.out.println(res);
```

This outputs:
```
26533.499999999996
265.33
```

I.e it rounds down not up as 26533.5 cannot be represented exactly as a double.

Instead a DecimalFormat or BigDecimal can be used to give correct rounding.

Please see the discussion here for more details https://stackoverflow.com/questions/153724/how-to-round-a-number-to-n-decimal-places-in-java

Reply:
I think this can also be done by accounting for the error in the double. Most likely this would be faster than using BigDecimal or DecimalFormat. Something like: 
```
public static double round(double num, int decimalPlaces) {

    int pow10 = 1;
    for (int i = 0; i < decimalPlaces; i++) {
      pow10 *= 10;
    }

    double dPow10 = pow10;

    double top = num * dPow10;

    double ulp = Math.ulp(top);
    
    return  Math.round(top + ulp) / dPow10;
    
  }
```
Would need testing, esp. with edge cases, e.g. probably doesn't work with -ve num and -ve decimal places as is.

Reply:
Having said that, probably the safest thing to do is to use BigDecimal :)